osbs/build/spec.py

```diff
@@ -165,7 +165,6 @@ def __init__(self):
             self.git_uri,
             self.git_ref,
             self.user,
-            self.component,
```

Comment:
Actually looking further at this test, I see it's whole purpose is to test whether deprecated options cause problems. :)

So I was wrong, we should include `self.component` here. In any case, we can't take it out altogether because it is a positional argument.